boskos/client/client.go

```diff
@@ -0,0 +1,224 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"k8s.io/test-infra/boskos/common"
+)
+
+var (
+	defaultURL = "http://boskos"
+)
+
+// Public Boskos client object
+type Client struct {
+	url       string
+	resources []string
```

Comment:
I'm not sure I understand why we need this / what it is doing that's valuable.

I guess this is to support the ReleaseAll and UpdateAll? How are you planning to use this?

Also have you considered making the Client auto-update on a timer and/or separate go func of sorts? This might be useful. Otherwise Kubetest needs to do this itself... On the other hand maybe we want Kubetest to do this since it is easy for it to do so.